src/redprl/signature.sml

```diff
@@ -474,11 +469,7 @@ struct
 
 
     fun insertAstDecl sign opid (decl, pos) =
```

Comment:
I think that the reason for doing this was in case the declaration already existed; if so, replacing it would invalidate the subsequent declarations. This would never come up in the current interaction model, though...

Is this change in response to observed behavior, or did the code just look weird?

Reply:
Just weird-looking. Since there is a `handle` for duplicate entries, I assumed that raising an error was the desired behavior. The reason that said error does actually get raised with the current code is because of (what I perceive to be) a bug in sml-telescopes, where `truncateFrom sign opid` doesn't actually remove the `opid` entry from the telescope's dictionary, despite removing the label from its list of labels.